### DISCUSSION TITLE
Unable to run Serverpod Generate after latest 1.1.0 upgrade

### DISCUSSION THREAD

**Discussion by lastmeta** [serverpod/serverpod#919] [Q&A] [answered]
We were running serverpod using version `0.9.21` for a while. Today we decided to upgrade to the `1.1.0` release.

After the `pubspec.yaml` files were updated, and got, and the code was all updated for `1.1.0`, I ran the following on my windows machine: 
```
C:\repos\project\project_server> serverpod generate
Can't load Kernel binary: Invalid kernel binary format version.
Could not find a file named "pubspec.yaml" in "C:\Users\...\AppData\Local\Pub\Cache\hosted\pub.dev\serverpod_cli-0.9.21".
```

I went to the folder (`C:\Users\...\AppData\Local\Pub\Cache\hosted\pub.dev\`) and noticed `serverpod_cli` package wasn't even installed. (maybe updating to `1.1.0` uninstalls it? idk).

so I then included `serverpod_cli: ^1.1.0` in the `pubspec.yaml`, still got the error.

so I then included `serverpod_cli: ^0.9.21` in the `pubspec.yaml`, just to see if that would help. But couldn't pull it because of this error:

```
C:\moontree\project\project_server> dart pub get
Resolving dependencies... (1.0s)
Because serverpod_cli >=0.9.21 <0.9.22 depends on
  serverpod_shared ^0.9.21 and serverpod_cli >=0.9.22 <1.0.0       
  depends on serverpod_service_client ^0.9.22, serverpod_cli       
  >=0.9.21 <1.0.0 requires serverpod_shared ^0.9.21 or
  serverpod_service_client ^0.9.22.
And because serverpod 1.1.0 depends on serverpod_shared ^1.1.0,    
  if serverpod_cli >=0.9.21 <1.0.0 and serverpod 1.1.0 then        
  serverpod_service_client ^0.9.22.
And because serverpod 1.1.0 depends on serverpod_service_client    
  ^1.1.0 and no versions of serverpod match >1.1.0 <2.0.0,
  serverpod_cli >=0.9.21 <1.0.0 is incompatible with serverpod     
  ^1.1.0.
So, because serverv2_server depends on both serverpod ^1.1.0 and   
  serverpod_cli ^0.9.21, version solving failed.
```

it seems to me that the `serverpod generate` binary/code is referencing a previous version rather than the current one (at least on Windows). Could this be the case? If not is there a problem with my particular installation?

Thanks!

**Comment by andriyndev** [comment]
Did you upgrade version in all three pubspec.yaml files (server, client, flutter)? If yes, the problem might also be if your project uses a library that is also used in Serverpod and the major versions of the library is different (I have had the problem because my project used `image 3.*.*` while Serverpod used `image 4.*.*`).

**Reply by lastmeta** [reply]
I upgraded all of them. I have other projects which are still on older versions, but I've never noticed their existence interfering in this manner in the past. I was able to set all the `pubspect.yaml`s to version 1.1.0 successfully.

**Reply by yahu1031** [reply]
You need to upgrade cli too.

**Reply by lastmeta** [reply]
you mean via the pubspec.yaml?

**Comment by lastmeta** [comment, marked as answer]
I updated my flutter and dart 
then ran:
`dart pub global activate serverpod_cli`
and it seems to be working great now.
